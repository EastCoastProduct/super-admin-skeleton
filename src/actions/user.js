import { USER_FETCH_SUCCESS, USER_GET_FAILED } from '../constants/actions';
import { API_URL } from '../constants/application';
import createFormData from '../utils/createFormData';
import parseErrors from '../utils/parseErrors';
import fetch from '../utils/fetch';

export const userFetchSuccess = profile => ({
  type: USER_FETCH_SUCCESS,
  profile,
});

export const userGetFailed = error => ({
  type: USER_GET_FAILED,
  error,
});

export const userGetFetch = userId =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`).then(resp =>
      dispatch(userFetchSuccess(resp))
    ).catch(err =>
      dispatch(userGetFailed(err.message))
    );

export const userUpdateFetch = (values, userId, cb) =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`, {
      method: 'POST',
      body: createFormData(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb();
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );

export const userCreateFetch = (values, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users`, {
      method: 'POST',
      body: JSON.stringify(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb(resp.id);
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );
