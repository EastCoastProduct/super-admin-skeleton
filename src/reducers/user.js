import { fromJS } from 'immutable';
import { USER_FETCH_SUCCESS, USER_GET_FAILED, USER_CHANGE_SUCCESS }
  from '../constants/actions';

const initialState = fromJS({
  error: null,
  profile: {},
});

export default (state = initialState, action) => {
  switch (action.type) {
    case USER_FETCH_SUCCESS:
      return state.merge(fromJS({
        error: null,
        profile: action.profile,
      }));
    case USER_CHANGE_SUCCESS:
      return state.merge(fromJS({
        error: null,
        profile: state.get('profile').merge(...action.responses),
      }));
    case USER_GET_FAILED:
      return state.merge(fromJS({
        error: action.error,
      }));
    default:
      return state;
  }
};
