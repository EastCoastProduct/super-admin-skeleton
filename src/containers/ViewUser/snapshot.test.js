import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import { fullProfile, profile } from '../../fixtures/user';
import { ViewUserComponent } from './';

describe('Users component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders empty required data', () => {
    const tree = renderer.create(
      <ViewUserComponent
        dispatch={() => {}}
        params={{ userId: profile.id }}
        profile={fromJS(profile)}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders full required data', () => {
    const tree = renderer.create(
      <ViewUserComponent
        dispatch={() => {}}
        params={{ userId: fullProfile.id }}
        profile={fromJS(fullProfile)}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
