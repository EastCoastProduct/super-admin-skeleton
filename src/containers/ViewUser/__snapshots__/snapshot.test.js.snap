exports[`Users component snapshot renders empty required data 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    View Profile
  </h1>
  <section
    className="box_5m47zw">
    <dl>
      <dt
        className="term_1hjxy0q">
        Email
      </dt>
      <dd
        className="data_g380iq">
        test@email.com
      </dd>
    </dl>
    <a
      className="link_1jhb6p"
      onClick={[Function]}
      style={Object {}}>
      Edit
    </a>
  </section>
</main>
`;

exports[`Users component snapshot renders full required data 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    John Doe
  </h1>
  <section
    className="box_5m47zw">
    <dl>
      <p>
        <div
          className="emptyImage_avz902"
          style={
            Object {
              "background": "url(https://s3.amazonaws.com/ecp-boilerplate/images/9049ffe0-b70f-11e6-92c0-51aa519864b8.jpg) center/cover no-repeat",
            }
          } />
      </p>
      <dt
        className="term_1hjxy0q">
        Email
      </dt>
      <dd
        className="data_g380iq">
        test@email.com
      </dd>
      <dt
        className="term_1hjxy0q">
        First Name
      </dt>
      <dd
        className="data_g380iq">
        John
      </dd>
      <dt
        className="term_1hjxy0q">
        Last Name
      </dt>
      <dd
        className="data_g380iq">
        Doe
      </dd>
      <dt
        className="term_1hjxy0q">
        Bio
      </dt>
      <dd
        className="data_g380iq">
        This is my bio.
      </dd>
    </dl>
    <a
      className="link_1jhb6p"
      onClick={[Function]}
      style={Object {}}>
      Edit
    </a>
  </section>
</main>
`;
